Week3/homework/6-step3.js

```diff
@@ -4,3 +4,12 @@ const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
 
 // add your solution here, or add a comment on how you would tackle this problem
+
+//I would tackle this problem by using a forEach() method, which execute a callback function for every element of the array, this way I could print how many nested elements there are. 
+
+function matrix(arr) {
+    arr.forEach(function(a, b) {
+      console.log(a, b);
+    });
+  }
+  matrix(arr3d);
```

Comment:
Hi @SignosRk - I am not sure if this is the correct way to solve this problem because forEach iterates over individual array element and applies the callback function. For example, in our case, it would call console log only twice and print `[[1, 2], [3, 4]]` & `[[5, 6], [7, 8]]` which I don't think is what we want.

This would be best handled by a pattern in programming which we call `recursion`. I would suggest that you give it a read on the internet. Briefly, recursion means calling the same function over and over again until it satisfies a condition.

Reply:
@yash-kapila ... Yes you are right, I didn't had that concept clear... that's why I solved this in a wrong way. I will investigate a bit more and practice more about the recursions. I've had too much information to process in javascript and not much time to practice everything, I knew this sunday about those functions calling itself xD. I will practice more in this topic man.. thanks for the feedback!

Reply:
Let me know if you have any questions around it.